phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/ui/SearchBox.java

```diff
@@ -1,31 +1,34 @@
 package org.phoenicis.javafx.views.mainwindow.ui;
 
-import javafx.event.ActionEvent;
-import javafx.event.EventHandler;
-import org.phoenicis.javafx.views.common.ThemeManager;
-
 import javafx.beans.property.StringProperty;
-import javafx.scene.control.TextField;
 import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
 import javafx.scene.layout.AnchorPane;
 
 import java.util.function.Consumer;
 
 public class SearchBox extends AnchorPane {
     private TextField searchField;
     private Button clearButton;
+    private Consumer<String> onSearch;
```

Comment:
There may be cases where we simply don't want (or even are unable) to implement a `onSearch` function. In these cases I would like accept an empty `onSearch` function. 

To allow an empty `onSearch` function we have two choices:

- the approach you've choosen, where you allow `null`
- an `Optional` approach, which doesn't require your `if ... else ...` block